.github/workflows/ci.yml

```diff
@@ -92,23 +93,20 @@ jobs:
           cache: "pip"
           cache-dependency-path: "setup.py"
 
-      - name: Upgrade pip and virtualenv to latest
-        run: pip install --upgrade pip virtualenv
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
 
       - uses: stripe/openapi/actions/stripe-mock@master
 
       - name: Typecheck with pyright
-        run: make pyright
-        # Pip won't install newer versions of pyright on python 3.6 (odd, because pyright is a node module)
-        # so we skip running pyright, and do double duty on python 3.7 using the --pythonversion flag.
-        if: matrix.python-version != '3.6'
-
-      - name: Typecheck with pyright
-        run: make pyright PYRIGHT_ARGS="--pythonversion 3.6"
-        if: matrix.python-version == '3.7'
+        run: PYRIGHT_ARGS="-- --pythonversion ${{ matrix.python-version }}" make pyright
+        # Skip typecheking in pypy legs
+        if: ${{ !contains(matrix.python-version, 'pypy') }}
 
       - name: Test with pytest
-        run: make ci-test
+        run: TOX_ARGS="-e ${{ matrix.python-version }}" make ci-test
```

Comment:
Looking at some of the cI output:

https://github.com/stripe/stripe-python/actions/runs/6882099159/job/18719881529?pr=1134#step:7:28

I'm not actually convinced that `pytest` is being run with different python interpreters -- it looks like 3.10 is running the test suite always? Or at least, there are a lot of mentions of 3.10

<img width="561" alt="image" src="https://github.com/stripe/stripe-python/assets/52928443/f721071c-c830-4aa9-9f96-7c40fec76bfe">


Can you double check this and/or possible cause pytest to output something that unambiguously indicates what python version it is running under?

Reply:
Fixed